From 570ac474d374e2c748e1a60eb5579e46ad1d125b Mon Sep 17 00:00:00 2001
From: Mike Drob <mdrob@apache.org>
Date: Fri, 9 Jun 2017 11:44:02 -0500
Subject: [PATCH] HBASE-18033 Fix license check for hadoop-3.x

Hadoop 3.x added some new dependencies that refer to the apache license
by it's older title (from apache parent pom 14 and earlier). Instead of
listing them all individually, we can group them under ASL for our
LICENSE file.
---
 .../src/main/resources/META-INF/LICENSE.vm                  |  2 +-
 .../src/main/resources/supplemental-models.xml              | 13 +++++++++++++
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/hbase-resource-bundle/src/main/resources/META-INF/LICENSE.vm b/hbase-resource-bundle/src/main/resources/META-INF/LICENSE.vm
index 96f6bceff0..e593f126ef 100644
--- a/hbase-resource-bundle/src/main/resources/META-INF/LICENSE.vm
+++ b/hbase-resource-bundle/src/main/resources/META-INF/LICENSE.vm
@@ -1550,7 +1550,7 @@ ${dep.scm.url}
 #thrift_license()
 #end
 ## Note that this will fail the build if we don't have a license. update supplemental-models.
-#if( !(${dep.licenses[0].name.contains("Apache License, Version 2.0")}) )
+#if( !(${dep.licenses[0].name.contains("Apache License, Version 2.0")} || ${dep.licenses[0].name.contains("The Apache Software License, Version 2.0")}) )
 #if( ${dep.licenses[0].name.contains("CDDL")} )
 #if( ${dep.licenses[0].name.contains("1.0")} )
 #set($aggregated = $cddl_1_0.add($dep))
diff --git a/hbase-resource-bundle/src/main/resources/supplemental-models.xml b/hbase-resource-bundle/src/main/resources/supplemental-models.xml
index 57ce56fa0c..5808d3c896 100644
--- a/hbase-resource-bundle/src/main/resources/supplemental-models.xml
+++ b/hbase-resource-bundle/src/main/resources/supplemental-models.xml
@@ -2272,6 +2272,19 @@ Copyright 2005 Sun Microsystems, Inc. and portions Copyright Apache Software Fou
   </supplement>
   <supplement>
     <project>
+      <groupId>javax.ws.rs</groupId>
+      <artifactId>jsr311-api</artifactId>
+      <licenses>
+        <license>
+          <name>Common Development and Distribution License (CDDL) v1.0</name>
+          <url>http://www.opensource.org/licenses/cddl1.php</url>
+          <distribution>repo</distribution>
+        </license>
+      </licenses>
+    </project>
+  </supplement>
+  <supplement>
+    <project>
       <groupId>org.jamon</groupId>
       <artifactId>jamon-runtime</artifactId>
       <name>Jamon runtime support classes</name>
-- 
2.13.0

